vcd/resource_vcd_org_vdc.go

```diff
@@ -536,6 +600,152 @@ func resourceVcdVdcDelete(d *schema.ResourceData, meta interface{}) error {
+// we approach this with adding new policies, set new default, remove all old policies.
+func changeVmSizingPoliciesAndDefaultId(d *schema.ResourceData, vcdComputePolicyHref string, vdc *govcd.AdminVdc) error {
+	if d.HasChange("default_vm_sizing_policy_id") && d.HasChange("vm_sizing_policy_ids") {
+		vmSizingPolicyIdStrings := convertSchemaSetToSliceOfStrings(d.Get("vm_sizing_policy_ids").(*schema.Set))
+		policyReferences := types.VdcComputePolicyReferences{}
+		var vdcComputePolicyReferenceList []*types.Reference
+		for _, policyId := range vmSizingPolicyIdStrings {
+			vdcComputePolicyReferenceList = append(vdcComputePolicyReferenceList, &types.Reference{HREF: vcdComputePolicyHref + policyId})
+		}
+
+		existingPolicies, err := vdc.GetAllAssignedVdcComputePolicies(nil)
+		if err != nil {
+			return fmt.Errorf("error getting VM sizing policies. %s", err)
+		}
+		for _, existingPolicy := range existingPolicies {
+			vdcComputePolicyReferenceList = append(vdcComputePolicyReferenceList, &types.Reference{HREF: vcdComputePolicyHref + existingPolicy.VdcComputePolicy.ID})
+		}
+
+		policyReferences.VdcComputePolicyReference = vdcComputePolicyReferenceList
+
+		_, err = vdc.SetAssignedComputePolicies(policyReferences)
+		if err != nil {
+			return fmt.Errorf("error setting VM sizing policies. %s", err)
+		}
+
+		// set default VM sizing policy
+		defaultPolicyId := d.Get("default_vm_sizing_policy_id").(string)
+		vdc.AdminVdc.DefaultComputePolicy = &types.Reference{HREF: vcdComputePolicyHref + defaultPolicyId, ID: defaultPolicyId}
+		updatedVdc, err := vdc.Update()
+		if err != nil {
+			return fmt.Errorf("error setting default VM sizing policy. %s", err)
+		}
+
+		// Now we can remove previously existing polices as default policy changed
+		vdcComputePolicyReferenceList = []*types.Reference{}
+		for _, policyId := range vmSizingPolicyIdStrings {
+			vdcComputePolicyReferenceList = append(vdcComputePolicyReferenceList, &types.Reference{HREF: vcdComputePolicyHref + policyId})
+		}
+		policyReferences.VdcComputePolicyReference = vdcComputePolicyReferenceList
+
+		_, err = updatedVdc.SetAssignedComputePolicies(policyReferences)
+		if err != nil {
+			return fmt.Errorf("error setting VM sizing policies. %s", err)
+		}
+	}
+	return nil
+}
+
+// addAssignedVmSizingPolicies handles VM sizing policies. Created VDC generates default VM sizing policy which requires additional handling.
+// Assigning and setting default VM sizing policies requires different API calls. Default approach is add new policies, set new default, remove all policies.
+func addAssignedVmSizingPolicies(vcdClient *VCDClient, d *schema.ResourceData, meta interface{}) error {
+
+	if vcdClient.Client.APIVCDMaxVersionIs(">= 33.0") {
+		log.Printf("[TRACE] updating assigned VM sizing policies to VDC")
+		_, idsOk := d.GetOk("vm_sizing_policy_ids")
+		_, defaultIdOk := d.GetOk("default_vm_sizing_policy_id")
+		if idsOk != defaultIdOk {
```

Comment:
Again - this will continue if both are `false`. Is this fine?

Reply:
look ok, cause later it uses d.HasChange to use values or not

Reply:
Yes, but if it does the same - but code readability would be a lot better if one can safely assume it is a no-op here

Reply:
add early return